src/app/tasks/services/tasks-filters.service.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { TaskStatus } from '@aneoconsultingfr/armonik.api.angular';
+import { TestBed } from '@angular/core/testing';
+import { FiltersAnd } from '@app/types/filters';
+import { DefaultConfigService } from '@services/default-config.service';
+import { TableService } from '@services/table.service';
+import { TasksFiltersService } from './tasks-filters.service';
+import { TasksStatusesService } from './tasks-status.service';
+import { TaskFilterDefinition, TaskSummaryFiltersOr } from '../types';
+
+
+describe('TasksFilterService', () => {
+  let service: TasksFiltersService;
+  const mockDefaultConfigService = {
+    defaultTasks: {
+      filters: [] as unknown as TaskSummaryFiltersOr 
+    }
+  };
+  const mockTasksStatusesService = {
+    statuses: {
+      0: $localize`Unspecified`,
+      3: $localize`Dispatched`,
+      1: $localize`Creating`,
+      2: $localize`Submitted`,
+      9: $localize`Processing`,
+      10: $localize`Processed`,
+      7: $localize`Cancelling`,
+      8: $localize`Cancelled`,
+      4: $localize`Finished`,
+      5: $localize`Error`,
+      6: $localize`Timeout`,
+      11: $localize`Retried`,
+    }
+  };
+  const mockTableService = {
+    saveFilters: jest.fn(),
+    resetFilters: jest.fn(),
+    restoreFilters: jest.fn(),
+  };
+  const mockFiltersDefinitions : TaskFilterDefinition[] = [
+    {
+      for: 'root',
+      field: 16,
+      type: 'string',
+    },
+    {
+      for: 'root',
+      field: 1,
+      type: 'string',
+    },
+    {
+      for: 'root',
+      field: 10,
+      type: 'string',
+    },
+    {
+      for: 'root',
+      field: 2,
+      type: 'status',
+      statuses: Object.keys(mockTasksStatusesService.statuses).map(status => {
+        return {
+          key: status,
+          value: mockTasksStatusesService.statuses[Number(status) as TaskStatus],
+        };
+      }),
+    },
+    {
+      for: 'options',
+      field: 5,
+      type: 'string',
+    },
+    {
+      for: 'options',
+      field: 6,
+      type: 'string',
+    },
+    {
+      for: 'options',
+      field: 4,
+      type: 'string',
+    }
+  ];
+
+  const filterAnd: FiltersAnd<number, number> = [{
+    field: 1,
+    for: 'root',
+    operator: 1,
+    value: 'Dummy'
+  }];
+  const mockTaskSummaryfilterOr: TaskSummaryFiltersOr = [
+    filterAnd
+  ];
+
+  beforeEach(() => {
+    service = TestBed.configureTestingModule({
+      providers: [
+        TasksFiltersService,
+        {provide: TasksStatusesService, useValue: mockTasksStatusesService},
+        {provide: DefaultConfigService, useValue: mockDefaultConfigService }, 
```

Comment:
I don't think DefaultConfigService needs to be mocked. It does not have any dependencies and its functions are quite simple. Mocking it is not a good way to work with this service.

You can just provide it naturally, and check that the values returned by its functions are correct !

```suggestion
    DefaultConfigService,
```